compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1761,47 +1761,51 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
             self.get_impl_future_output_ty(exp_found.expected),
             self.get_impl_future_output_ty(exp_found.found),
         ) {
-            (Some(exp), Some(found)) if ty::TyS::same_type(exp, found) => match &cause.code {
-                ObligationCauseCode::IfExpression(box IfExpressionCause { then, .. }) => {
-                    diag.multipart_suggestion(
-                        "consider `await`ing on both `Future`s",
-                        vec![
-                            (then.shrink_to_hi(), ".await".to_string()),
-                            (exp_span.shrink_to_hi(), ".await".to_string()),
-                        ],
-                        Applicability::MaybeIncorrect,
-                    );
-                }
-                ObligationCauseCode::MatchExpressionArm(box MatchExpressionArmCause {
-                    prior_arms,
-                    ..
-                }) => {
-                    if let [.., arm_span] = &prior_arms[..] {
+            (Some(exp), Some(found)) if ty::TyS::same_type_modulo_infer(exp, found) => {
```

Comment:
What a strange diff. This didn't do anything except for replace `same_type` with `same_type_modulo_infer` and run the autoformatter, which caused an extra block to get added in.

Reply:
If you look at it [hiding whitespace changes, the diff looks more reasonable](https://github.com/rust-lang/rust/pull/91022/files?diff=split&w=1). This happens because the line went over 100 columns, so it decided to reformat the arm into a block. It's fine.